src/openrct2-ui/windows/Map.cpp

```diff
@@ -764,9 +765,12 @@ static Widget window_map_widgets[] = {
         {
             int32_t x = 0, y = 0, dx = 0, dy = 0;
 
-            int32_t pos = (_currentLine * (getMiniMapWidth() - 1)) + getTechnicalMapSize() - 1;
-            auto destinationPosition = ScreenCoordsXY{ pos % getMiniMapWidth(), pos / getMiniMapWidth() };
-            auto destination = _mapImageData.data() + (destinationPosition.y * getMiniMapWidth()) + destinationPosition.x;
+            auto miniMapWidth = getMiniMapWidth();
+            int32_t pos = (_currentLine * (miniMapWidth - 1)) + getPracticalMapSize() - 1;
+            auto destinationPosition = ScreenCoordsXY(0, 0);
+            if (miniMapWidth > 0)
```

Comment:
Hmm, I don't like the added guard here, to be honest. I'd also prefer not to have to add one to `getPracticalMapSize` either. 

Frankly, I think the hacked map should be rejected as corrupted instead. At best, I think the map size should be corrected to a positive integer when the park is imported, but honestly, there's no saving it, anyway, is there?